helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataWriter.java

```diff
@@ -77,7 +88,23 @@ public synchronized boolean addMetadataStoreRealm(String realm) {
       return createZkRealm(realm);
     }
 
-    // TODO: Forward the request to leader
+    String leaderHostName = _leaderElection.getCurrentLeaderInfo().getId();
+    String url =
+        leaderHostName + "/admin/v2/namespaces/" + _namespace + "/metadata-store-realms/" + realm;
+    HttpPut putRequest = new HttpPut(url);
+    try {
+      HttpResponse response = _forwardHttpClient.execute(putRequest);
```

Comment:
Could we configure a timeout for these requests? 

https://www.baeldung.com/httpclient-timeout

Something like 1 minute sounds reasonable. Make that 1 minute a constant somewhere and you could use it throughout.

Reply:
I'll take a look. Very good point.

Reply:
@narendly What's your opinion on the necessity of setting a socket timeout in addition to a connection timeout?